ZelBack/src/services/fluxCommunication.js

```diff
@@ -386,6 +386,86 @@ async function handleAppRemovedMessage(message, fromIP, port) {
   }
 }
 
+async function handleNodeDownMessage(message, fromIP, port) {
+  const nodeConnectivityService = require('./nodeConnectivityService');
+
+  try {
+    const { ip: downNodeIP, broadcastAt } = message.data;
+
+    if (!downNodeIP || typeof downNodeIP !== 'string') {
+      log.warn(`handleNodeDownMessage: Invalid nodedown message from ${fromIP}:${port} - missing or invalid IP`);
+      return;
+    }
+
+    // Check if message is less than 5 minutes old
+    const currentTime = Date.now();
+    const messageAge = currentTime - broadcastAt;
+    const fiveMinutes = 5 * 60 * 1000; // 5 minutes in milliseconds
+
+    if (!broadcastAt || messageAge > fiveMinutes) {
+      log.warn(`handleNodeDownMessage: Ignoring old nodedown message from ${fromIP}:${port} - message age: ${messageAge}ms`);
```

Comment:
The message is discarded if older than 5 min, but is later rebroadcast only when verifyTimestampInFluxBroadcast passes a 4 min(240000) window. A message between those is accepted yet never rebroadcast.